Pagination/MongoTrait/FilterTrait.php

```diff
@@ -138,22 +139,42 @@ protected function generateFilterSearchField($name, $value, $type)
     }
 
     /**
-     * @param Stage       $qa
      * @param array|null  $order
      * @param array|null  $descriptionEntity
      *
      * @return Stage
      */
-    protected function generateFilterSort(Stage $qa, $order = null , $descriptionEntity = null)
+    protected function generateArrayForFilterSort($order = null , $descriptionEntity = null, $returnOrder = true)
     {
         if (null !== $order) {
             $columnsName = $order['name'];
-            if (isset($descriptionEntity[$columnsName]) && isset($descriptionEntity[$columnsName]['field'])) {
-                $name = $descriptionEntity[$columnsName]['field'];
-                $dir = ($order['dir'] == 'desc') ? -1 : 1;
-                $qa->sort(array($name => $dir));
+            if (isset($descriptionEntity[$columnsName])){
+                if($descriptionEntity[$columnsName] instanceof Search) {
+                    $name = $descriptionEntity[$columnsName]->getField();
+                } elseif (isset($descriptionEntity[$columnsName]['field'])) {
+                    $name = $descriptionEntity[$columnsName]['field'];
+                }
+                $dir = $name;
```

Comment:
in the case where $descriptionEntity[$columnsName] isn't an instance of Search and the key field not exists the parameter $name doesn't exist